.github/workflows/publish-dotnet.yaml

```diff
@@ -1,135 +1,77 @@
-
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Install dependencies for aarch64-unknown-linux-gnu
-        if: matrix.target == 'aarch64-unknown-linux-gnu'
-        run: sudo apt-get update && sudo apt-get install -y gcc-aarch64-linux-gnu
-
-      - name: Configure cargo for cross-compilation
-        if: matrix.target == 'aarch64-unknown-linux-gnu'
-        run: |
-          mkdir -p .cargo
-          echo '[target.aarch64-unknown-linux-gnu]' >> .cargo/config
-          echo 'linker = "aarch64-linux-gnu-gcc"' >> .cargo/config
-
-      - name: Install rust
-        run: |
-          rustup set auto-self-update disable
-          rustup toolchain install stable --profile default
-          rustup target add ${{ matrix.target }}
-          rustup show
-
-      - name: Rust cache
-        uses: Swatinem/rust-cache@v2
-
-      - name: Build yggdrasilffi
-        run: |
-          cargo build -p yggdrasilffi --release --target ${{ matrix.target }}
-
-      - name: Upload yggdrasilffi binary
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ matrix.rid }}
-          path: target/${{ matrix.target }}/release/${{ matrix.output }}
 
   publish:
-    needs: build-binaries
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: Download yggdrasilffi binaries
-        run: |
-          for rid in linux-x64 linux-arm64 win-x64 win-arm64 osx-x64 osx-arm64; do
-            mkdir -p dotnet-engine/runtimes/${rid}/native
-            echo "Downloading artifact for $rid"
-            if ! gh run download --name $rid --dir dotnet-engine/runtimes/${rid}/native; then
-              echo "Failed to download artifact for $rid, continuing..."
-            fi
-          done
-        env:
-          GH_TOKEN: ${{ github.token }}
-
       - name: Set up .NET
         uses: actions/setup-dotnet@v3
         with:
           dotnet-version: '6.0.x'
 
-      - name: Restore dependencies
+      - name: Extract Yggdrasil Core Version
+        id: extract_ffi_version
         run: |
-          cd dotnet-engine
-          dotnet restore dotnet-engine.sln
+          sudo apt-get install -y xmlstarlet
+          cd dotnet-engine/Yggdrasil.Engine
+          CORE_VERSION=$(xmlstarlet sel -t -v "/Project/PropertyGroup/YggdrasilCoreVersion" -n Yggdrasil.Engine.csproj)
+          echo "FFI Library Version: $CORE_VERSION"
+          echo "CORE_VERSION=$CORE_VERSION" >> $GITHUB_ENV
 
-      - name: Set version
+      - name: Download and Organize Binaries
+        env:
+          RELEASE_VERSION: ${{ github.event.inputs.release_version }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          cd dotnet-engine
-          version=${{ github.event.inputs.version }}
-          sed -i "s/<Version>.*<\/Version>/<Version>$version<\/Version>/" Yggdrasil.Engine/Yggdrasil.Engine.csproj
+          binaries=("libyggdrasilffi_aarch64.so linux/aarch64/libyggdrasilffi.so"
```

Comment:
Would love to have this list of binaries defined in one place across all builds but I can't see an obvious and sane way to do that right now

For the .NET case, we also have to put the binaries in specific locations, which makes this more challenging to abstract

Open to suggestions, but I'll probably do that in a future PR